aten/src/ATen/native/cuda/CuFFTPlanCache.h

```diff
@@ -15,51 +15,102 @@
 
 namespace at { namespace native { namespace detail {
 
-// This POD struct is used to let us easily compute hashes of the
+enum class CuFFTTransformType : int8_t {
+  C2C,  // Complex-to-complex
+  R2C,  // Real-to-complex
+  C2R,  // Complex-to-real
+};
+
+// This struct is used to let us easily compute hashes of the
 // parameters.
 // It will be the **key** to the plan cache.
 struct CuFFTParams
 {
-  at::ScalarType scalar_type_;
-  int64_t input_sizes_[max_rank + 2];
-  int64_t input_strides_[max_rank + 2];
-  uint8_t signal_ndim_;  // between 1 and max_rank, i.e., 1 <= signal_ndim <= 3
-  bool complex_input_;
-  bool complex_output_;
-  int64_t signal_sizes_[max_rank];
-  bool onesided_;
+  int64_t signal_ndim_; // between 1 and max_rank, i.e., 1 <= signal_ndim <= 3
+  // These include additional batch dimension as well.
+  int64_t sizes_[max_rank + 1];
+  int64_t input_strides_[max_rank + 1];
+  int64_t output_strides_[max_rank + 1];
+  CuFFTTransformType fft_type_;
+  ScalarType value_type_;
+
+  CuFFTParams() = default;
+
+  CuFFTParams(IntArrayRef in_strides, IntArrayRef out_strides,
+      IntArrayRef signal_sizes, CuFFTTransformType fft_type, ScalarType value_type) {
```

Comment:
Style nit: use a member initializer list and style the input names `_foo` so you have code like:

```
_foo) : foo_{_foo}, ...
```

Reply:
The member initializers would be overwritten by the the `memset` call which is necessary to zero the padding in the struct.

Reply:
Ah, OK. No worries then.